tasmota/berry/drivers/Core200S.be

```diff
@@ -0,0 +1,712 @@
+            return self.set_display_auto_off_mode(DisplayAutoOffMode.ON)
+        end
+    end
+
+    def set_timer(seconds)
+        # check it timer if set and don't try to clean timer if not set
+        # (The stock firmware also checks if timer is set before it cleans it)
+        if seconds == 0 && self.timer_total == 0
+            return true
+        end
+        var msg = bytes("a522080800210164a20000000000")
+        msg.set(10, seconds, 3)
+        return ser.send(msg) != bytes("")
+    end
+
+    def parse_status_message(msg)
+        var filter_reset_msg = bytes("a522070400250165a200")
+        if msg.size() == 22
+            self.fan_mode = FanMode(msg[13])
+            log_c2s("FanMode: " +  self.fan_mode.tostring())
+            self.sl_mode = SleepMode(msg[14])
+            log_c2s("SleepMode: " +  self.sl_mode.tostring())
+            self.fan_speed_mode = FanSpeedMode(msg[15])
+            log_c2s("FanSpeedMode: " +  self.fan_speed_mode.tostring())
+            self.dp_mode = DisplayMode(msg[16])
+            log_c2s("DisplayMode: " +  self.dp_mode.tostring())
+            self.dp_auto_off_mode = DisplayAutoOffMode(msg[17])
+            log_c2s("DisplayAutoOffMode: " +  self.dp_auto_off_mode.tostring())
+            self.cl_mode = ChildLockMode(msg[20])
+            log_c2s("ChildLockMode: " +  self.cl_mode.tostring())
+            self.nl_mode = NightLightMode(msg[21])
+            log_c2s("NightLightMode: " +  self.nl_mode.tostring())
+
+            # When timer is set it sends a status message (but not a message that actually contains the timer values)
+            # Thus on each status update we should query if a timer has been set
+            if msg[1] == 0x22
+                tasmota.delay(50)
+                self.query_timer()
+            end
+        elif filter_reset_msg == msg
+            info_c2s("Detected Filter Reset. Not Implemented...")
+        elif msg.size() == 18
+            self.timer_remaining = msg.get(10, 3)
+            self.timer_total = msg.get(14, 3)
+            log_c2s(format("TimerRemaining: %d", self.timer_remaining))
+            log_c2s(format("TimerTotal: %d", self.timer_total))
+        end
+
+
+    end
+
+    def every_250ms()
+        var msg = ser.read_msg(true)
+
+        while msg != bytes("")
+            info_c2s("Incoming Message " + msg.tohex())
+
+            if msg[0] == 0xa5
+                if msg[1] == 0x22
+                    ser.send_nowait(bytes("a512000400a301604000"))
+                    self.parse_status_message(msg)
+                elif msg[1] == 0x12
+                    self.parse_status_message(msg)
+                else
+                    log_c2s("Unrecognized message type")
+                end
+            end
+            msg = ser.read_msg(true)
+        end
+    end
+
+    def every_second()
+        self.second_counter += 1
+        if self.second_counter == 60
+            self.second_counter = 0
+            self.set_wifi_led(tasmota.wifi() != {})
```

Comment:
`tasmota.wifi()` is never equal to `{}`. What are you trying to do ? If you want to know if wifi is connected, use `tasmota.wifi('up')`

Reply:
ah, then i guess I misread the documentation (https://tasmota.github.io/docs/Berry/#tasmota-object) . It says 

>  Retrieves Wi-Fi connection info or empty map.

But i guess this means then that it only returns empty map if wifi is not supported (or maybe disabled) ? Maybe the documentation could be a bit more specific here...

Reply:
Good point. "empty map" is outdated, the object was changed later. I will update the documentation